src/gantt.ts

```diff
@@ -2454,26 +2633,47 @@ export class Gantt implements IVisual {
 
         taskRectMerged.classed(Gantt.TaskRect.className, true)
 
+        let index = 0, groupedTaskIndex = 0;
         taskRectMerged
             .attr("d", (task: Task) => this.drawTaskRect(task, taskConfigHeight, barsRoundedCorners))
             .attr("width", (task: Task) => this.getTaskRectWidth(task))
-            .style("fill", (task: Task) => task.color);
+            .style("fill", (task: Task) => {
+                // logic used for grouped tasks, when there are several bars related to one category
+                if (index === task.index) {
+                    groupedTaskIndex++;
+                } else {
+                    groupedTaskIndex = 0;
+                    index = task.index;
+                }
+
+                const url = `${task.index}-${groupedTaskIndex}-${isStringNotNullEmptyOrUndefined(task.taskType) ? task.taskType.toString() : "taskType"}`;
+                const encodedUrl = `task${hashCode(url)}`;
+
+                return `url(#${encodedUrl})`;
+            })
+            // stroke width is 0, so it will be invisible, but you'll be able to know the actual color
+            // instead of selecting <linearGradient> element and getting it's color
```

Comment:
why it is not good to take a color from an element? Not sure if it is a good idea to keep 0 stroke just for a color. Can you explain your idea?